riscv-page-tables/src/page_table.rs

```diff
@@ -465,6 +460,10 @@ pub trait PagingMode {
     /// Calculates the number of PTE pages that are needed to map all pages for `num_pages` mapped
     /// pages for this type of page table.
     fn max_pte_pages(num_pages: u64) -> u64;
+
+    /// Used to calculate the number of PTE pages needed if all the pages are
+    /// not 4k
+    fn max_huge_pte_pages(num_pages: u64, page_size: PageSize) -> u64;
```

Comment:
What will `max_huge_pte_pages` be used for? If used to minimize the number of pages needed, it would limit the ability to break huge pages to 4k. That might be awkward.